cmd/bosun/sched/template.go

```diff
@@ -383,3 +386,39 @@ func (c *Context) HTTPPost(url, bodyType, data string) string {
 	}
 	return string(body)
 }
+
+//func LSBaseQuery(e *State, index_root, keystring, filter, interval, sduration, eduration string) (*elastic.SearchService, *elastic.SearchSource, []keyMatch, error) {
+
+func (c *Context) LSContextQuery(index_root, filter, interval, sduration, eduration string) (interface{}, error) {
+	var ks []string
+	for k, v := range c.Group {
+		ks = append(ks, strings.Join([]string{k, v}, ":"))
+	}
+	fmt.Println(strings.Join(ks, ","))
+	return c.LSQuery(index_root, strings.Join(ks, ","), filter, interval, sduration, eduration)
+}
+
+func (c *Context) LSQuery(index_root, keystring, filter, interval, sduration, eduration string) (interface{}, error) {
+	//service, s, keys, err := expr.LSBaseQuery(c.State, index_root, keystring, filter, interval, sduration, eduration)
+
+	service, s, _, err := expr.LSBaseQuery(time.Now(), c.schedule.Conf.LogstashElasticHost, index_root, keystring, filter, interval, sduration, eduration)
+	if err != nil {
+		return nil, err
+	}
+	results, err := service.SearchSource(s).Do()
+	if err != nil {
+		return nil, err
+	}
+	r := make([]*simplejson.Json, len(results.Hits.Hits))
+	for i, h := range results.Hits.Hits {
+		s := string([]byte(*h.Source)[:])
+		var err error
+		r[i], err = simplejson.Loads(s)
```

Comment:
It makes for a good template interface for accessing members of the returned document. For example:

```
	{{$ls := .LSContextQuery .Alert.Vars.index_root "" .Alert.Vars.interval .Alert.Vars.start ""}}
	<table>
	{{range $h := $ls}}
	<pre>{{$h}}</pre>
	<tr>
		<td>{{$h.Get "TargetUserName"}}</td>
		<td>{{($h.Get "IpAddress").String}}</td>
	</tr>
	{{end}}
	</table>
```